shared/src/backend/fetch.ts

```diff
@@ -1,18 +1,30 @@
 import { isErrorLike } from '../util/errors'
 
-export const EHTTPSTATUS = 'HTTPStatusError'
+const EHTTPSTATUS = 'HTTPStatusError'
 
 export class HTTPStatusError extends Error {
     public readonly name = EHTTPSTATUS
     public readonly code = EHTTPSTATUS
-    constructor(public response: Response) {
+    public readonly status: number
+
+    constructor(response: Response) {
         super(`Request to ${response.url} failed with ${response.status} ${response.statusText}`)
+        this.status = response.status
     }
 }
-export const isHTTPStatusError = (err: unknown): err is HTTPStatusError => isErrorLike(err) && err.code === EHTTPSTATUS
 
 /**
- * Checks if a given fetch Response has a HTTP 2xx status code and throws an Error otherwise.
+ * Lenient helper to check if an error was a HTTPStatusError that failed with a specific HTTP status.
+ * Works also if the error was sent from a web worker or background page (see https://github.com/mozilla/webextension-polyfill/issues/210).
+ *
+ * @param error The error
+ * @param status The status code to check
+ */
+export const failedWithHTTPStatus = (error: unknown, status: number): boolean =>
+    isErrorLike(error) && error.message.includes(` failed with ${status} `)
```

Comment:
Should this optimistically handle the case where the error is well-formed? It would be in the webapp, and in native integrations.

```suggestion
    isHTTPStatusError(error) ? error.response.status === 401 : isErrorLike(error) && error.message.includes(` failed with ${status} `)
```

Reply:
I deleted `isHTTPStatusError()` for the time being so that it is not used instead of this function. It could check both conditions of course, but I don't really see a benefit of that? Might be faster in theory but I don't think it's measurable. A counter point is it may make breakages harder to detect if e.g. the error message is changed in the class but not here

Reply:
The benefit I would see is to not check against error messages (which isn't ideal) when we can avoid it. But I don't feel strongly about it and don't see it as blocking this PR.